dpmData.py

```diff
@@ -28,38 +24,52 @@ def main():
     args = parser.parse_args()
     sys.stdout.write(str(hdf_code(args)))
 
+def local_to_utc(date):
+    utc_timezone = datetime.timezone(datetime.timedelta(0))
+    utc_datetime_obj = date.astimezone(utc_timezone)
+    time_in_ms = int(utc_datetime_obj.timestamp() * 1000)
+    return time_in_ms
+
 def hdf_code(args):
     START_DATE = args.start_date
     END_DATE = args.end_date
     DURATION = args.duration
     DEVICE_LIMIT = args.device_limit
     DEVICE_FILE = args.device_file
     OUTPUT_FILE = args.output_file
-    #NODE = args.node #unused
+    NODE = args.node
 
+    request_string = ''     #Used later to provide input string for DPM
+    
     if DURATION and (START_DATE or END_DATE):
         print("-d and -s|-e are mutually exclusive! Exiting ...")
         sys.exit(2)
+
     elif(START_DATE and END_DATE):
-        DURATION = END_DATE - START_DATE
-        #print(int((END_DATE - START_DATE).total_seconds()))
-        DURATION = str(DURATION)
-        DURATION = 'LOGGERDURATION:' + DURATION
-    elif(DURATION):
-        #Transform duration from int to str, and then add "LOGGERDURATION:" before it.
+        time_window = 'LOGGER:' + str(local_to_utc(START_DATE)) + ':' + str(local_to_utc(END_DATE))
+        request_string = time_window
+
+    elif DURATION:
+        DURATION *= 1000
         DURATION = str(DURATION)
         DURATION = 'LOGGERDURATION:' + DURATION
-
+        request_string = DURATION
```

Comment:
This line is unnecessary. Move the assignment up a line. `request_string = 'LOGGERDURATION:' + DURATION`.
My preference would be `request_string = f'LOGGERDURATION:{str(DURATION * 1000)}`, but that's just my preference. This is functionally the same.

Reply:
I tried request_string = f'LOGGERDURATION:{str(DURATION * 1000)} but that just gives me an error. So I have used request_string = 'LOGGERDURATION:' + str(DURATION) instead.